src/viam/components/base/client.py

```diff
@@ -130,3 +130,11 @@ async def do_command(
         request = DoCommandRequest(name=self.name, command=dict_to_struct(command))
         response: DoCommandResponse = await self.client.DoCommand(request, timeout=timeout)
         return struct_to_dict(response.result)
+
+    async def get_geometries(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None) -> List[Geometry]:
+        if extra is None:
+            extra = {}
+        request = GetGeometriesRequest(name=self.name, extra=dict_to_struct(extra))
+        response: GetGeometriesResponse = await self.client.GetGeometries(request, timeout=timeout)
+        geometries = [geometry for geometry in response.geometries]
+        return geometries
```

Comment:
consider refactoring this out into a separate function

Reply:
in this case I meant the entire four lines or so (like in https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/utils.ts#L36-L58)